README.md

```diff
@@ -108,43 +108,94 @@ See [CONFIG.md](CONFIG.md) for details.
 See [Code Samples](https://github.com/hazelcast/hazelcast-nodejs-client/tree/master/code_samples)
 
 
-# Serialization Considerations
+# Serialization
 
-Hazelcast needs to serialize objects in order to be able to keep them in the server memory. For primitive types, it uses Hazelcast native serialization. For other complex types (e.g. JS objects), it uses JSON serialization.
+Serialization is the process of converting an object into a stream of bytes to store the object in memory, a file or database, or transmit it through network. Its main purpose is to save the state of an object in order to be able to recreate it when needed. The reverse process is called deserialization. Hazelcast offers you its own native serialization types. You will see these types throughout the chapter.
 
-For example, when you try to query your data using predicates, this querying is handled on the server side so Hazelcast does not have to bring all data to the client but only the relevant entries. Otherwise, there would be a lot of unneccessary data traffic between the client and the server and the performance would severely drop.
-Because predicates run on the server side, the server should be able to reason about your objects. That is why you need to implement serialization on the server side.
+Hazelcast serializes all your objects before sending them to the server. For the following types Hazelcast optimizes the serialization and you cannot override this behavior.
 
-The same applies to MapStore. The server should be able to deserialize your objects in order to store them in MapStore.
+- A `boolean` type which is a primitive on the Node.js client can be serialized as a `Boolean` for Java server side.
+- A character which is a single-length string one can be serialized as a `Character` for Java server side.
+- A `number` type which is a primitive on the Node.js client can be serialized as `Byte`, `Short`, `Integer`, `Float`, `Double` for Java server side.
+- A `Long` type which represents a 64 bit two's complement integer value can be serialized as a `Long` for Java server side.
+- A `string` type which is a primitive on the Node.js client can be serialized as a `String` for Java server side.
+- Arrays of the above types can be serialized as `boolean[]`, `char[]`, `byte[]`, `short[]`, `int[]`, `float[]`, `double[]`, `long[]` and `string[]` for Java server side respectively. 
 
-Regarding arrays in a serializable object, you can use methods like `writeIntArray` if the array is of a primitive type.
+Note that if the object is not one of the above-mentioned types, the Node.js client uses `JSON Serialization` by default. You can create your complex object and it is serialized by using `JSON Serialization`. After serialization, the byte array format of your object is sent to the server through network.
```

Comment:
I think this part does not provide any valuable information.
```
You can create your complex object and it is serialized by using `JSON Serialization`. After serialization, the byte array format of your object is sent to the server through network.
```
We can remove that part keeping only the first sentence and merge the next paragraph to this one.